sensor/kubernetes/complianceoperator/handler_impl.go

```diff
@@ -325,9 +348,13 @@ func (m *handlerImpl) processRerunScheduledScanRequest(requestID string, request
 	// created from a scan configuration have the same name as scan
 	// configuration.
 	resI := m.client.Resource(complianceoperator.ComplianceSuite.GroupVersionResource()).Namespace(ns)
-	obj, err := resI.Get(m.ctx(), request.GetScanName(), v1.GetOptions{})
+	var obj *unstructured.Unstructured
+	err := m.callWithRetry(func(ctx context.Context) error {
+		var err error
+		obj, err = resI.Get(ctx, request.GetScanName(), v1.GetOptions{})
+		return errors.Wrapf(err, "namespaces/%s/compliancesuites/%s not found", ns, request.GetScanName())
+	})
 	if err != nil || obj == nil {
```

Comment:
We are currently not returning an error if `obj == nil` and `err == nil` - is that intended?

Reply:
tbh. I'm not sure. Case to get `nil` for both is not easy to understand, and it's going down to the decoder on the rest api. If we want to cover that, I would do it in a follow-up.

My idea was to have something like:
```
err = errors.Wrapf(err, "namespaces/%s/compliancescans/%s not found", ns, scan.Name)
```
as before, but where `errors.Wrapf` will return an error even when `err` is nil.

If we want to be more specific, then we can split it into two `if`s.

**EDIT:** Maybe it would be good to check how we are handling that in other places where the Kubernetes client is used.